sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java

```diff
@@ -465,7 +466,7 @@ private static DefaultVolumeSpec convertVolume(
                 rawVolume.getSize(),
                 volumeTypeEnum,
                 rawVolume.getPath(),
-                role,
+                ResourceUtils.getRole(preReservedRole, role),
```

Comment:
Yes, I've been thinking how I can make sure I'm not missing a spot.  I think I have a way, testing it out now.